tests/Functional/FileSyncer/FileSyncerTestCase.php

```diff
@@ -0,0 +1,158 @@
+        self::deleteFile($stagingDir, 'delete_from_staging_dir_before_syncing_back_to_active_dir.txt');
+        self::createFile($stagingDir, 'create_in_staging_dir.txt');
+
+        $sut = $this->createSut();
+        $sut->sync($stagingDir, $activeDir, $exclusions);
+
+        self::assertDirectoryListing($activeDir, [
+            'no_changes_ever.txt', // Unchanged file was preserved and unsupported absolute path exclusion was ignored.
+            'exclude_dir/to_test_recursion.txt', // Did not delete from the active directory an excluded directory absent from the staging directory.
+            'exclude_dir/and_change_file_in_active_dir_after_syncing_to_staging_dir.txt',
+            'exclude_file.txt', // Did not delete from the active directory an excluded file absent from the staging directory.
+            'change_in_staging_dir_before_syncing_back_to_active_dir.txt',
+            // This file should be absent from the active directory because it was deleted from the staging directory:
+            // delete_from_staging_dir_before_syncing_back_to_active_dir.txt
+            'very/deeply/nested/file/that/is/never/changed/in/either/the/active/directory/or/the/staging/directory.txt', // Correctly handled deeply nested, unchanged file.
+            'very/deeply/nested/file/that/is/changed/in/the/staging/directory/before/syncing/back/to/the/active/directory.txt', // Correctly handled deeply nested, changed file.
+            'long_filename_lorem_ipsum_dolor_sit_amet_consectetur_adipiscing_elit_sed_do_eiusmod_tempor_incididunt_ut_labore_et.txt', // Correctly handled long filename with weird characters.
+            'exclude_long_filename_lorem_ipsum_dolor_sit_amet_consectetur_adipiscing_elit_sed_do_eiusmod_tempor_incididunt_ut_labore_et.txt', // Correctly excluded long filename with weird characters.
+            'create_in_staging_dir.txt', // Copied back to the active directory a new file that was created new in the staging directory.
+        ], $stagingDir);
+
+        self::assertFileNotChanged($activeDir, 'no_changes_ever.txt', 'Preserved a preexisting file in the active directory that was never changed anywhere.');
+        self::assertFileChanged($activeDir, 'very/deeply/nested/file/that/is/changed/in/the/staging/directory/before/syncing/back/to/the/active/directory.txt', 'Preserved a preexisting file in the active directory that was never changed anywhere.');
+        self::assertFileChanged($activeDir, 'exclude_dir/and_change_file_in_active_dir_after_syncing_to_staging_dir.txt', 'Preserved a preexisting file in the active directory that was never changed anywhere.');
+        self::assertFileChanged($activeDir, 'change_in_staging_dir_before_syncing_back_to_active_dir.txt', 'Preserved in the active directory changes made to a file in the staging directory.');
+    }
+
+    public function providerDirectories(): array
+    {
+        return [
+            // Siblings without trailing slashes.
```

Comment:
These comments describing the scenarios might be better repurposed as the keys of the scenario array, for easier inspection if they fail. Example:

```
return [
  'siblings without trailing slashes' => [
    'activeDir' => 'active-dir',
    'stagingDir' => 'staging-dir',
  ],
  // ...
];
```

Reply:
Yeah, let's. Done.